### ISSUE TITLE
ASM Multi-Project Multi-Cluster Setup

### ISSUE THREAD

**Issue by pb250134** [terraform-google-modules/terraform-google-kubernetes-engine#1520]
### TL;DR

We have a handful of applications, which few are dependent on a few key applications, and these are deployed in different clusters in different projects. 
We wanted to bring them into one umbrella using Anthos Service Mesh, even though they still be in different clusters in different projects. By making the Mesh/Fleet project common for all these projects and clusters, in which all these clusters in different projects use the same fleet ID so that the monitoring of these fall under one.
We wanted to achieve this using the terraform code, where instead of instrumenting the ASM module for each cluster once, we wanted to use the same module where we can pass the cluster/project/other required details as a list or map in the variables to that module so that it can run through and make use of one fleet project ID to register for all the required clusters.

### Terraform Resources

```markdown
https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/tree/master/modules/asm
```

### Detailed design

```markdown
Right now we are using the below code block to instrument ASM resources on the cluster using terraform, which will be duplicated for all the required clusters and executed sequentially to achieve the required.


module "asm" {
  source                    = "terraform-google-modules/kubernetes-engine/google//modules/asm"
  project_id                = var.project
  cluster_name              = module.gke.name
  cluster_location          = module.gke.location
  multicluster_mode         = "connected"
  enable_cni                = true
  enable_fleet_registration = true
  enable_mesh_feature       = false
  internal_ip               = true
  channel                   = "stable"
  fleet_id                  = var.network_project_id
}
```

Google team confirms [1] that Multi-Cluster Multi-Project is supported for ASM installation, we would like to know how the same can be achieved using the terraform ASM module [2] in one go.

[1] https://cloud.google.com/service-mesh/docs/unified-install/gke-install-multi-cluster
[2] https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/tree/master/modules/asm
```


### Additional information

_No response_
